src/Functions/getSetting.cpp

```diff
@@ -58,17 +61,51 @@ class FunctionGetSetting : public IFunction, WithContext
             throw Exception(ErrorCodes::ILLEGAL_COLUMN,
                             "The argument of function {} should be a constant string with the name of a setting",
                             String{name});
+        if (arguments.size() == 2)
```

Comment:
You should check for mode as well

Reply:
Done, thanks! Checked for mode now (2 arguments will be present if mode is Default)